### PR TITLE
Do not allow checking of mesh properties without a type

### LINKED ISSUE #23643: Do not allow checking of mesh properties without a type
## Reason
The mesh property system is type driven. It does not make sense to allow querying it without type.

## Design
Remove the ability to query mesh properties without a type.

## Impact
Locks down the mesh property API, hopefully leading to less heartache in the future.

### PR DESCRIPTION
**Pull request by loganharbour**
Closes https://github.com/idaholab/moose/issues/23643

### CODE REVIEW

**Comment by loganharbour** [comment]
FYI @miaoyinb - this removes the ability to call `hasMeshGenerator()` without a type. 

Having this feature doesn't make sense because everything is typed, plus it vastly complicated the new automatic-property-from-parent capabilities

**Comment by loganharbour** [comment]
@GiudGiud when you have a moment please. trying to keep them simple - this closes an API that is difficult to maintain

File: modules/reactor/src/meshgenerators/PatternedCartesianPeripheralModifier.C [hunk 1]
```diff
@@ -32,7 +32,7 @@ PatternedCartesianPeripheralModifier::PatternedCartesianPeripheralModifier(
 std::unique_ptr<MeshBase>
 PatternedCartesianPeripheralModifier::generate()
 {
-  if (hasMeshProperty("square_center_trimmability", _input_name))
+  if (hasMeshProperty<Real>("square_center_trimmability", _input_name))
```
Comment (GiudGiud) [inline modules/reactor/src/meshgenerators/PatternedCartesianPeripheralModifier.C:35]: bool ?
why isnt it erroring?

**Review by GiudGiud — CHANGES_REQUESTED** [review]

**Comment by miaoyinb** [comment]
> FYI @miaoyinb - this removes the ability to call `hasMeshGenerator()` without a type.
> 
> Having this feature doesn't make sense because everything is typed, plus it vastly complicated the new automatic-property-from-parent capabilities

Thanks. Just curious what would happen if a wrong type is used here, like `uint32_t` instead of `uint16_t`.

**Comment by loganharbour** [comment]
> Thanks. Just curious what would happen if a wrong type is used here, like `uint32_t` instead of `uint16_t`.

The types need to be exact.

File: modules/reactor/src/meshgenerators/PatternedCartesianPeripheralModifier.C (hunk 1 above)
Comment (loganharbour) [reply modules/reactor/src/meshgenerators/PatternedCartesianPeripheralModifier.C]: This is good now, should have been bool.

**Comment by loganharbour** [comment]
@miaoyinb would you mind looking at the reactor changes to make sure that the types are sane?

**Comment by miaoyinb** [comment]
Looks good to me. Thanks.

**Review by GiudGiud — APPROVED** [review]

**Comment by loganharbour** [comment]
Failure is unrelated.

**Pull request merged**